diff --git a/index.bs b/index.bs
--- index.bs
+++ index.bs
@@ -11385,15 +11385,15 @@
 <pre class="cddl remote-cddl local-cddl">
 webExtension.Extension = text
 </pre>

-The <code>webExtension.Extension</code> type represents a web extension id within a browser.
+The <code>webExtension.Extension</code> type represents a web extension id within a [=remote end=].

 ### Commands ### {#module-webExtension-commands}

 #### The webExtension.install Command #### {#command-webExtension-install}

-The <dfn export for=commands>webExtension.install</dfn> command installs a web extension for the browser.
+The <dfn export for=commands>webExtension.install</dfn> command installs a web extension in the [=remote end=].

 <dl>
    <dt>Command Type</dt>
    <dd>
@@ -11462,20 +11462,25 @@
    <dl>
     <dt>|type| is the string "<code>path</code>"
     <dd>
       1. Let |path| be |extension data spec|["<code>path</code>"].
-      1. Let |entry| be a [=directory entry=] representing the directory with that |path|.
+      1. Let |locator| be a [=directory locator=] with [=file system locator/path=] |archive path| and [=file system locator/root=] corresponding to the root of the file system.
+      1. Let |entry| be [=locate an entry=] given |locator|.

     <dt>|type| is the string "<code>archivePath</code>"
     <dd>
       1. Let |archive path| be |extension data spec|["<code>path</code>"].
-      1. Perform implementation defined steps to read |bytes| from a file located at |archive path|.
+      1. Let |locator| be a [=file locator=] with [=file system locator/path=] |archive path| and [=file system locator/root=] corresponding to the root of the file system.
+      1. Let |archive entry| be [=locate an entry=] given |locator|.
+      1. If |archive entry| is null, return null.
+      1. Let |bytes| be |archive entry|'s [=file entry/binary data=].
       1. Let |entry| be the result of [=trying=] to [=extract a zip archive=] given |bytes|.

     <dt>|type| is the string "<code>base64</code>"
     <dd>
       1. Let |bytes| be [=forgiving-base64 decode=] |extension data spec|["<code>path</code>"].
       1. Let |entry| be the result of [=trying=] to [=extract a zip archive=] given |bytes|.
+      1. If |entry| is failure, return null.

    </dl>

 1. Return |entry|.
@@ -11490,8 +11495,10 @@
 1. Let |extension data spec| be |command parameters|["<code>extensionData</code>"].

 1. Let |extension directory entry| be the result of [=trying=] to [=expand a web extension data spec=] with |extension data spec|.

+1. If extension directory entry| is null, return [=error=] with [=error code=] [=invalid web extension=].
+
 1. Perform implementation defined steps to install a web extension from |extension directory entry|. If this fails, return [=error=] with [=error code=] [=invalid web extension=]. Otherwise let |extension id| be the unique identifier of the newly installed web extension.

 1. Let |result| be a [=/map=] matching the
    <code>webExtension.InstallResult</code> production with the
@@ -11526,11 +11533,11 @@
 The [=remote end steps=] with |command parameters| are:

 1. Let |extension| be |command parameters|["<code>extension</code>"].

-1. If the browser has no web extension with an id equal to |extension|, return [=error=] with [=error code=] [=no such web extension=].
+1. If the [=local end=] has no web extension with id equal to |extension|, return [=error=] with [=error code=] [=no such web extension=].

-1. Perform any implementation-defined steps to remove the web extension from the browser. If this fails, return [=error=] with [=error code=] [=unknown error=].
+1. Perform any implementation-defined steps to remove the web extension from the [=local end=]. If this fails, return [=error=] with [=error code=] [=unknown error=].

 1. Return [=success=] with data null.

 </div>
